CMakeLists.txt

```diff
@@ -224,6 +224,40 @@ if(WITH_FAISS_TESTS)
   add_subdirectory(tests/faiss)
 endif()
 
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+
+    # Find ARM SVE headers
+    find_path(ARM_SVE_DIR arm_sve.h PATHS
+        /usr/lib/gcc/aarch64-linux-gnu/*/include
+        /usr/lib/llvm-*/lib/clang/*/include
+        /usr/include
+        /usr/local/include
+        NO_DEFAULT_PATH
+    )
+    if(ARM_SVE_DIR)
+        include_directories(SYSTEM ${ARM_SVE_DIR})
+        message(STATUS "ARM SVE headers found at: ${ARM_SVE_DIR}")
+    else()
+        message(WARNING "ARM SVE headers not found!")
+    endif()
+
+    # Find ARM NEON headers
+    find_path(ARM_NEON_DIR arm_neon.h PATHS
+        /usr/lib/gcc/aarch64-linux-gnu/*/include
+        /usr/lib/llvm-*/lib/clang/*/include
+        /usr/include
+        /usr/local/include
+        NO_DEFAULT_PATH
+    )
+    if(ARM_NEON_DIR)
+        include_directories(SYSTEM ${ARM_NEON_DIR})
+        message(STATUS "ARM NEON headers found at: ${ARM_NEON_DIR}")
+    else()
+        message(WARNING "ARM NEON headers not found!")
+    endif()
+
+endif()
+
```

Comment:
delete those lines, compiler know where to find arm_neon.h

Reply:
have tried, can build pass without this file change

Reply:
@Presburger ,

Initially, my plan was to keep SVE support optional until all the functions were implemented. For this reason, I provided the -march=armv8-a+sve flag as an optional tag for users to add. However, as per the discussion with @alexanderguzhva , we have now hardcoded -march=armv8-a+sve in the faiss CMake file. Consequently, I agree that we can remove the optional flag from CMake.

This approach works well after removing the optional tag. However, I had intentionally added those lines earlier because I believed the CI setup was incomplete. The reason for this was that not all users rely directly on the Docker environment provided by the CI. Some build the project manually, and if their machine has a different GCC version and they haven’t properly updated their alternatives, it could have caused issues.

While the CI has now been fixed, future developers may still encounter problems in similar situations as and when they build it on local machine. We can consider a better long-term solution to address this later on.